### PR TITLE
[WIP] feat: prepare test command

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR adds `prepare-test` command. It takes inputs of Profile examples and prepares test file which can be used with our testing library. Missing more tests, some more logs and readme. I will add it if we decide this command makes sense.

**NOTE: this is more like POC right now*
*

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/logic/prepare-test.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  options?: {
+    logCb?: LogCallback;
+  }
+): Promise<void> {
+  const profileFiles = await loadProfile(superJson, profile, version, options);
+  const parameters: Record<string, ExampleInput[]> = {};
+
+  for (const usecase of profileFiles.ast.definitions) {
+    if (isUseCaseDefinitionNode(usecase)) {
+      const examples = usecase.examples || [];
```
Comment (martinalbert) [inline src/logic/prepare-test.ts:160]:
```suggestion
  const {
    ast: { definitions },
  } = await loadProfile(superJson, profile, version, options);
  const usecases = definitions.filter(isUseCaseDefinitionNode);
  const parameters: Record<string, ExampleInput[]> = {};

  for (const usecase of usecases) {
    const examples = usecase.examples || [];
```

File: src/templates/testTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  let body = '';
+
+  for (const parameter of Object.keys(parameters)) {
+    body += testUseCaseTemplete(parameter, parameters[parameter]);
+  }
+
+  return `${head}
+  ${body}\n
+});`;
```
Comment (martinalbert) [inline src/templates/testTemplate.ts:30]: Extra line here.

```suggestion
  return `${head}
  ${body}
});`;
```

**Review by martinalbert — CHANGES_REQUESTED** [review]

File: src/logic/prepare-test.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+ * @returns
+ */
+function intend(numberOfTabs: number): string {
+  let res = '';
+  for (let i = 0; i < numberOfTabs; i++) {
+    res += '\t';
+  }
+
+  return res;
+}
```
Comment (lukas-valenta) [inline src/logic/prepare-test.ts:35]: first, why tabs? :scream: 
second: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat

Comment (TheEdward162) [reply src/logic/prepare-test.ts:35]: Yees, let the tabs flow through you 😈

File: src/logic/prepare-test.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+/**
+ * Prepares ExampleInput object
+ * @param input UseCaseExampleNode to be used as an input
+ * @returns undefined if node does not contain any examples
+ */
+function prepareExampleInput(
+  input: UseCaseExampleNode
+): ExampleInput | undefined {
+  if (input.input === undefined || input.exampleName === undefined) {
```
[the remark this answers is not in the document]
Comment (Jakub-Vacek) [reply src/logic/prepare-test.ts:110]: I wasn't sure if we want to reuse that name, so 🤷🏻‍♂️ I will look into it more.

**Comment by Jakub-Vacek** [comment]
Closed in favor of https://github.com/superfaceai/cli/pull/234

**Pull request closed without merging**